### PR TITLE
Use repository API for merges when possible

### PR DESCRIPTION
**Pull request by sstok**
| Q | A |
| --- | --- |
| Fixed tickets | #584 |
| License | MIT |

Work in progress, needs testing

### CODE REVIEW

**Comment by sstok** [comment]
OK, it works but it includes the `Merge pull request #nn from sstok/xxxx` header.
So that's something that needs work.

**Comment by cordoval** [comment]
that does not need work, that is just ok, if user decided to merge it via the cloud is what it is supposed to say. It is ok right?

**Comment by sstok** [comment]
It is fixable using the `polaris-preview` https://developer.github.com/v3/pulls/#merge-a-pull-request-merge-button

You can merge this as-is now, so that I can fix the merge message tomorrow 👍

**Comment by sstok** [comment]
polaris-preview is not going to work, the Knp GitHub library doesn't support the `commit_title` yet and the method (`put()`) to do this manually is protected. Instead I added an option to allow disabling the merge-api.

**Comment by cordoval** [comment]
@phansys please rebase PR

File: src/Command/PullRequest/PullRequestMergeCommand.php [hunk 1]
```diff
@@ -225,6 +231,44 @@ protected function execute(InputInterface $input, OutputInterface $output)
         }
     }
 
+    private function canUseApi(InputInterface $input)
+    {
+        if (true === $this->getConfig()->get('disable_merge_api', Config::CONFIG_ALL, false)) {
+            return false;
+        }
+
+        if ($input->getOption('squash') ||
```
Comment (GrahamCampbell) [inline src/Command/PullRequest/PullRequestMergeCommand.php]: You can still squash commits using the preview github API.

Comment (sstok) [inline src/Command/PullRequest/PullRequestMergeCommand.php]: True, but that will complicate the Adapter API 😄

Comment (GrahamCampbell) [inline src/Command/PullRequest/PullRequestMergeCommand.php]: Not too much though. Probably worth it.

Comment (sstok) [inline src/Command/PullRequest/PullRequestMergeCommand.php]: Any suggestions? You need to detect whether the Adapter support this operation.
But I don't like adding yet another interface like `CanSquashPullRequestOnMerge` 💀 

Adding an extra arguments on the method is no problem btw 👍

**Pull request closed without merging**